api/attestation/attestation.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package attestation
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/edgelesssys/ego/attestation"
+	"github.com/edgelesssys/ego/attestation/tcbstatus"
+)
+
+// TCBStatusError is returned when the TCB status of a Coordinator enclave is not accepted by a given configuration.
+type TCBStatusError struct {
+	// TCBStatus is the TCB status of the Coordinator enclave.
+	TCBStatus tcbstatus.Status
+}
+
+// NewTCBStatusError creates a new TCBStatusError.
+func NewTCBStatusError(tcbStatus tcbstatus.Status) error {
+	return &TCBStatusError{TCBStatus: tcbStatus}
+}
+
+// Error returns the error message.
+func (e *TCBStatusError) Error() string {
+	return fmt.Sprintf("invalid TCB status: %s", e.TCBStatus)
+}
+
+// Config is the expected attestation metadata of a MarbleRun Coordinator enclave.
+// It is used to verify the Coordinator's remote attestation report.
+// At minimum, either UniqueID or the tuple of SignerID, ProductID, and SecurityVersion must be provided.
+type Config struct {
+	SecurityVersion     uint
+	UniqueID            string
+	SignerID            string
+	ProductID           uint16
+	Debug               bool
+	Nonce               []byte
+	AcceptedTCBStatuses []string
+}
+
+// VerifyCertificate verifies the Coordinator's TLS certificate against the Coordinator's SGX quote.
+// A config with the expected attestation metadata must be provided.
+// It returns the TCB status of the enclave, the quote, and an error, if any.
+func VerifyCertificate(certificate *pem.Block, quote []byte, config Config) error {
+	report, err := verifyRemoteReport(quote)
+	if err != nil && !errors.Is(err, attestation.ErrTCBLevelInvalid) {
+		return err
+	}
+
+	if err := verifyReport(report, certificate.Bytes, config); err != nil {
+		return err
+	}
+
+	// Check if the TCB status is accepted
+	if !slices.Contains(config.AcceptedTCBStatuses, report.TCBStatus.String()) {
+		return NewTCBStatusError(report.TCBStatus)
```

Comment:
This is a behavior change for the CLI: UpToDate was always accepted, even if not in AcceptedTCBStatuses.
Also, the messages "TCB level accepted by configuration" and "Warning: TCB level invalid, but accepted by configuration" are gone.

Reply:
I could re-add usage of the `/internal/tcb` package here.
Though I would prefer to not print anything when just using the API outside of the CLI.
Maybe a global `SetLogger` function could solve the issue.
By default nothing is logged, and the CLI can use the function to instruct the API to log this information to stdout